docs/types/dynamic-dispatch.md

```diff
@@ -163,16 +163,32 @@ When invoking a method on _module object_ its _module static methods_ take
 precedence over _instance methods_ defined on `Any`. Thus a module serves
 primarily as a _container for module (static) methods_.
 
-## Method Resolution
+## Method invocation
 
 **Terminology**:
 
 - **eigen type** of a type `My_Type` is a type of type, usually written as
   `My_Type.type`.
   - Every type has an eigen type.
   - Eigen type of an eigen type is itself.
+  - There is always exactly one instance of each eigen type, that is, eigen type
+    is singleton itself.
+  - The name is motivated by an
+    [eigen value of a matrix](https://en.wikipedia.org/wiki/Eigenvalues_and_eigenvectors).
 - **associated type** of a module `My_Module` is a type for the module
   - It is basically an eigen type for a module.
+- **builtin type** is a type annotated with `@Builtin_Type`.
```

Comment:
- relevant question is: [What is a benefit of builtin type](https://github.com/enso-org/enso/issues/13664#issuecomment-3266102569)?
- unless I am mistaken we could easily run without `@Builtin_Type`
- I mean builtin type - not `@Builtin_Method` - that's still very much needed
- on and on, I'd rather leave _builtin types_ out of the method resolution document
- they don't have any special behavior to discuss - or do they?

Reply:
> relevant question is: https://github.com/enso-org/enso/issues/13664#issuecomment-3266102569?

That is indeed very relevant.

> they don't have any special behavior to discuss - or do they?

The only reason why I am mentioning builtin types in terminology section here, is because only builtin types may have different inheritance hierarchy. An example is later mentioned in [parent type](https://github.com/enso-org/enso/blob/41914f42ec66ae4b64c8b10215b79619e8bce9e3/docs/types/dynamic-dispatch.md#L189-L191).